### DISCUSSION TITLE
Page selection: Exclude pages instead of include? (for example --pages x5,x17,x23)

### DISCUSSION THREAD

**Discussion by amishmm** [qpdf/qpdf#563] [Q&A] [answered]
I have a big PDF file with say 100 pages.

I do not want some random pages in it (say page 5,17,23,29,75 so on)

So for page selection I have to do --pages in.pdf 1-4,6-16,18-22,24-28,30-74,76-so on.

Is there an easier way where I can specify pages to exclude?

Like --pages in.pdf x5,x17,x23,x29,x75 so on? This will select all pages except page 5,17,23,29,75

If feature is not there then I would like to propose the feature.

Feature can be extended as a range say, x5-10 i.e. exclude pages 5 to 10. Keep rest.

There can be more combination but not going in detail currently.

Thank you.

**Comment by jberkenbilt** [comment]
Feel free to propose the feature as an issue. I'll consider it, at least for the library API. The `--pages` command line syntax is already complicated and causes a lot of confusion, but the syntax you suggest would not conflict with any existing syntax and may be possible for me to add. If you're interested in proposing this, please create an issue. You can do that with the "Create issue from discussion" link on the right side of the page.

**Comment by amishmm** [comment]
Link to the issue created:
https://github.com/qpdf/qpdf/issues/564

**Comment by jberkenbilt** [comment, marked as answer]
This will be in 11.7.1.
